design/history/exploration-reports/2021.01-encryption.md

```diff
@@ -0,0 +1,224 @@
+Encryption and IPLD, 2021
+=========================
+
+This is an exploration report about the role and relationship of encryption relating to IPLD,
+gathering some thoughts and recent updates in early 2021.
+It's meant to be useful as a reference piece for further discussion at this time.
+
+This document takes input from tons of people.
+It's written by warpfork in the immediate aftermath of close conversation with Mikeal,
+but also has tons of input from Carson of Textile (via the 2021.01.11 IPLD Weekly meeting),
+and also draws on other notes exchanged over time with project such as the Ceramic Network, 3Box, Peergos, Qri, and others.
+(Even if you don't see your name here, it's likely you've contributed something --
+this topic has just been a long time brewing, so attributing all inspirations completely is now hard!)
+Thank you to all these folks for their efforts.
+
+
+Overview
+--------
+
+People frequently want to implement encryption as part of decentralized systems.
+So, it's no surprise that it's also frequent that people want a story for how encryption and IPLD should interact.
+
+For a long time, IPLD has been agnostic to any sort of encryption.
+(We've been afraid of doing a _wrong_ thing and baking it into specs.)
+Instead, we've asked that people using IPLD figure out how to compose IPLD and encryption on their own.
+It may now be time for this to change, as we gather lots of input from the community.
+
+In this document, we're going to cover three major topic groups:
+
+- 1. A proposal for encryption primitives in IPLD, and a plan for how to use multicodec indicators for encryption!
+- 2. A section about usage conventions we see which have repeatedly emerged, and seem useful, and thus now seem worth identifying and creating vocabulary for.
+- 3. A section for gathering notes about use cases, tradeoff notes, and general cautions about general applied cryptography.
+
+Comments and feedback on each of these topic groups are welcome.
+
+
+Encryption Codecs
+-----------------
+
+The IPLD team is now ready to plan (and implement) encryption which is signaled by multicodec indicators
+(and thus works anywhere CIDs are used),
+and works in the natural way an IPLD codec is expected to work.
+
+(This is a big change in stance.
+Previously, we've considered it unclear whether codecs)
+
+There's a couple of details about how we expect this to work which are recent realizations,
+and so this document might be nearly the first description of them:
+
+
+### encryption codecs use multicodec indicators
```

Comment:
this section might want to be updated - the current approach is to have a single new `ipld` codec which encodes the cipher as a multicodec integer, so we have one new codec (for now) and a bunch of ciphers in the table.